​MailKit​ generates error message for changes in the user profile (9.9.0 RC1, MailKit vs CoreMailProvider)

<!-- 
  If you need community support or would like to solicit a Request for Comments (RFC), please post to the DNN Community forums at https://dnncommunity.org/forums for now.  In the future, we are planning to implement a more robust solution for cultivating new ideas and nuturing these from concept to creation.  We will update this template when this solution is generally available.  In the meantime, we appreciate your patience as we endeavor to streamline our GitHub focus and efforts.
  
  Please read the CONTRIBUTING guidelines at https://github.com/dnnsoftware/Dnn.Platform/blob/develop/CONTRIBUTING.md prior to submitting an issue.

  Any potential security issues SHOULD NOT be posted on GitHub.  Instead, please send an email to security@dnnsoftware.com.
-->
## Description of bug
​MailKit​ generates error message
CoreMailProvider function

## Steps to reproduce

​MailKit​
SMTP server settings:
Server name and port:: mail.1234.ch:465 send via SSL: yes, sent with no problem
Profile Settings:
go to the profile settings and change entries there, click save... nothing happens
go to Admin Logs:
02/11/2021 21:56:59 Page Load Exception hans.meier Testportal AbsoluteURL /Default.aspx DefaultDataProvider DotNetNuke.Data.SqlDataProvider
AbsoluteURL:/Default.aspx
DefaultDataProvider:DotNetNuke.Data.SqlDataProvider, DotNetNuke
ExceptionGUID:197513c3-5854-4908-a66c-34b3fceecd57
AssemblyVersion:9.9.0
PortalId:0
UserId:1
TabId:22
RawUrl:/Activity-Feed/ctl/Profile/userId/1/pageno/2
Referrer:.../Activity-Feed/ctl/Profile/userId/1/pageno/2
UserAgent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15
ExceptionHash:cTQDKPsKmWPgWq1f6fAyi+UaB0g=
Message:Invalid addr-spec token at offset 0
StackTrace:
InnerMessage:Invalid addr-spec token at offset 0
InnerStackTrace:

   bei MimeKit.InternetAddress.TryParseAddrspec(Byte[] text, Int32& index, Int32 endIndex, Byte[] sentinels, Boolean throwOnError, String& addrspec, Int32& at)
   bei MimeKit.MailboxAddress.set_Address(String value)
   bei DotNetNuke.Services.Mail.MailKitMailProvider.SendMail(MailInfo mailInfo, SmtpInfo smtpInfo)
   bei DotNetNuke.Services.Mail.Mail.SendMail(Int32 portalId, Int32 userId, MessageType msgType, PortalSettings settings)
   bei DotNetNuke.Modules.Admin.Users.EditUser.ProfileUpdateCompleted(Object sender, EventArgs e)
   bei DotNetNuke.Entities.Modules.ProfileUserControlBase.OnProfileUpdateCompleted(EventArgs e)
   bei DesktopModules.Admin.Security.DNNProfile.cmdUpdate_Click(Object sender, EventArgs e)
   bei System.Web.UI.WebControls.LinkButton.OnClick(EventArgs e)
   bei System.Web.UI.WebControls.LinkButton.RaisePostBackEvent(String eventArgument)
   bei System.Web.UI.Page.<ProcessRequestMainAsync>d__523.MoveNext()
Source:
FileName:
FileLineNumber:0
FileColumnNumber:0
Method:
Server Name: Faulhorn


**CoreMailProvider** Works securely with ports (25), 587, 2525 TLS
No error in the event log

## Affected version
<!-- 
Please add X in at least one of the boxes as appropriate. In order for an issue to be accepted, a developer needs to be able to reproduce the issue on a currently supported version. If you are looking for a workaround for an issue with an older version, please visit the forums at https://dnncommunity.org/forums
-->
* [ ] 10.00.00 alpha build
* [x ] 09.09.00 release candidate
* [ ] 09.08.01 latest supported release

## Affected browser
<!-- 
  Check all that apply, and add more if necessary. As appropriate, please specify the exact version(s) of the browser and operating system.
-->
* [ ] Chrome
* [ ] Firefox
* [x ] Safari
* [ ] Internet Explorer 11
* [ ] Microsoft Edge (Classic)
* [ ] Microsoft Edge Chromium

Please excuse. Now I've loaded the Symbols package, but it no longer shows me an error in the event log at the moment.
I can only find these entries in the server log

2021-02-12 00:02:29.172+01:00 [Faulhorn][D:7][T:34][FATAL] DotNetNuke.Framework.PageBase - An error has occurred while loading page.
MimeKit.ParseException: Invalid addr-spec token at offset 0
   bei MimeKit.InternetAddress.TryParseAddrspec(Byte[] text, Int32& index, Int32 endIndex, Byte[] sentinels, Boolean throwOnError, String& addrspec, Int32& at)
   bei MimeKit.MailboxAddress.set_Address(String value)
   bei DotNetNuke.Services.Mail.MailKitMailProvider.SendMail(MailInfo mailInfo, SmtpInfo smtpInfo)
   bei DotNetNuke.Services.Mail.Mail.SendMail(Int32 portalId, Int32 userId, MessageType msgType, PortalSettings settings)
   bei DotNetNuke.Modules.Admin.Users.EditUser.ProfileUpdateCompleted(Object sender, EventArgs e)
   bei DotNetNuke.Entities.Modules.ProfileUserControlBase.OnProfileUpdateCompleted(EventArgs e)
   bei DesktopModules.Admin.Security.DNNProfile.cmdUpdate_Click(Object sender, EventArgs e)
   bei System.Web.UI.WebControls.LinkButton.OnClick(EventArgs e)
   bei System.Web.UI.WebControls.LinkButton.RaisePostBackEvent(String eventArgument)
   bei System.Web.UI.Page.<ProcessRequestMainAsync>d__523.MoveNext()
-----
2021-02-12 00:07:53.260+01:00 [Faulhorn][D:9][T:5][ERROR] DotNetNuke.Services.Exceptions.Exceptions - ~/Default.aspx?tabid=22&error=Invalid+addr-spec+token+at+offset+0
MimeKit.ParseException: Invalid addr-spec token at offset 0
   bei MimeKit.InternetAddress.TryParseAddrspec(Byte[] text, Int32& index, Int32 endIndex, Byte[] sentinels, Boolean throwOnError, String& addrspec, Int32& at) in D:\src\MimeKit\MimeKit\InternetAddress.cs:Zeile 384.
   bei MimeKit.MailboxAddress.set_Address(String value) in D:\src\MimeKit\MimeKit\MailboxAddress.cs:Zeile 264.
   bei DotNetNuke.Services.Mail.MailKitMailProvider.SendMail(MailInfo mailInfo, SmtpInfo smtpInfo) in D:\a\1\s\DNN Platform\Library\Services\Mail\MailKitMailProvider.cs:Zeile 47.
   bei DotNetNuke.Services.Mail.Mail.SendMail(Int32 portalId, Int32 userId, MessageType msgType, PortalSettings settings) in D:\a\1\s\DNN Platform\Library\Services\Mail\Mail.cs:Zeile 216.
   bei DotNetNuke.Modules.Admin.Users.EditUser.ProfileUpdateCompleted(Object sender, EventArgs e) in D:\a\1\s\DNN Platform\Website\DesktopModules\Admin\Security\EditUser.ascx.cs:Zeile 642.
   bei DotNetNuke.Entities.Modules.ProfileUserControlBase.OnProfileUpdateCompleted(EventArgs e) in D:\a\1\s\DNN Platform\Library\Entities\Modules\ProfileUserControlBase.cs:Zeile 36.
   bei DesktopModules.Admin.Security.DNNProfile.cmdUpdate_Click(Object sender, EventArgs e) in D:\a\1\s\DNN Platform\Website\DesktopModules\Admin\Security\Profile.ascx.cs:Zeile 220.
   bei System.Web.UI.WebControls.LinkButton.OnClick(EventArgs e)
   bei System.Web.UI.WebControls.LinkButton.RaisePostBackEvent(String eventArgument)
   bei System.Web.UI.Page.<ProcessRequestMainAsync>d__523.MoveNext()

--- a/DNN Platform/Library/Services/Mail/MailKitMailProvider.cs
+++ b/DNN Platform/Library/Services/Mail/MailKitMailProvider.cs
@@ -45,7 +45,8 @@ namespace DotNetNuke.Services.Mail
             }
 
             var mailMessage = new MimeMessage();
-            mailMessage.From.Add(new MailboxAddress(mailInfo.FromName, mailInfo.From));
+
+            mailMessage.From.Add(ParseAddressWithDisplayName(displayName: mailInfo.FromName, address: mailInfo.From));
             if (!string.IsNullOrEmpty(mailInfo.Sender))
             {
                 mailMessage.Sender = MailboxAddress.Parse(mailInfo.Sender);
@@ -98,12 +99,14 @@ namespace DotNetNuke.Services.Mail
 
                     if (needUpdateSender)
                     {
-                        mailMessage.Sender = new MailboxAddress(senderDisplayName, senderAddress);
+                        mailMessage.Sender = ParseAddressWithDisplayName(displayName: senderDisplayName, address: senderAddress);
                     }
                 }
                 else if (smtpInfo.Username.Contains("@"))
                 {
-                    mailMessage.Sender = new MailboxAddress(Host.SMTPPortalEnabled ? PortalSettings.Current.PortalName : Host.HostTitle, smtpInfo.Username);
+                    mailMessage.Sender = ParseAddressWithDisplayName(
+                        displayName: Host.SMTPPortalEnabled ? PortalSettings.Current.PortalName : Host.HostTitle,
+                        address: smtpInfo.Username);
                 }
             }
 
@@ -199,6 +202,17 @@ namespace DotNetNuke.Services.Mail
 
                 return retValue;
             }
+        }
+
+        private static MailboxAddress ParseAddressWithDisplayName(string displayName, string address)
+        {
+            var mailboxAddress = MailboxAddress.Parse(address);
+            if (!string.IsNullOrWhiteSpace(displayName))
+            {
+                mailboxAddress.Name = displayName;
+            }
+
+            return mailboxAddress;
         }
     }
 }

